### PR TITLE
Add a basic test of backwardLimitCompensation()

### PR DESCRIPTION
**Pull request by AndyZe**
I want to add some more tests of basic TrackJoint functionality. This is a first shot at that -- it tests jerk handling in backwardLimitCompensation().

I tried to add comments to explain this as best I can but I'll admit, the logic in this function is pretty complicated.

When merging, please squash the commits.

### CODE REVIEW

File: src/single_joint_generator.cpp [hunk 1]
```diff
@@ -216,6 +216,13 @@ Eigen::VectorXd SingleJointGenerator::interpolate(Eigen::VectorXd& times)
 
 ErrorCodeEnum SingleJointGenerator::forwardLimitCompensation(size_t* index_last_successful)
 {
+  // The algorithm:
+  // 1) check jerk limits, from beginning to end of trajectory. Don't bother
+  // checking accel/vel limits here, they will be checked next
+  // 2) check accel limits. Make sure it doesn't cause jerk to exceed limits.
+  // 3) check vel limits. This will also check whether previously-checked
+  // jerk/accel limits were exceeded
```
Comment (AndyZe) [inline src/single_joint_generator.cpp:224]: I moved this comment block up from L369. It was completely in the wrong place.

File: src/single_joint_generator.cpp [hunk 2]
```diff
@@ -366,29 +373,27 @@ ErrorCodeEnum SingleJointGenerator::forwardLimitCompensation(size_t* index_last_
...
-  // Add a bit of velocity at step i to compensate for the limit at timestep
-  // i+1.
-  // Cannot go beyond index 2 because we use a 2-index window for derivative
-  // calculations.
+  // Since (index + 1) is used in some calculations below, the algorithm won't work if
+  // limited_index is the last element in the array
+  if (limited_index == (unsigned) waypoints_.velocities.size() - 1)
+  {
+    return false;
+  }
```
Comment (AndyZe) [inline src/single_joint_generator.cpp:383]: Writing the test helped identify this edge case  :+1:

**Review by stephanie-eng — APPROVED** [review]
